vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java

```diff
@@ -0,0 +1,298 @@
+ *
+ * @author Vaadin Ltd.
+ */
+@Tag("vaadin-popover")
+@NpmPackage(value = "@vaadin/polymer-legacy-adapter", version = "24.5.0-alpha1")
+@NpmPackage(value = "@vaadin/popover", version = "24.5.0-alpha1")
+@JsModule("@vaadin/polymer-legacy-adapter/style-modules.js")
+@JsModule("@vaadin/popover/src/vaadin-popover.js")
+public class Popover extends Component implements HasComponents {
+
+    private Component target;
+    private Registration targetAttachRegistration;
+
+    /**
+     * Constructs an empty popover.
+     */
+    public Popover() {
+        getElement().getNode().addAttachListener(this::attachComponentRenderer);
+    }
+
+    /**
+     * Sets position of the popover with respect to its target.
+     *
+     * @param position
+     *            the position to set
+     */
+    public void setPosition(PopoverPosition position) {
+        getElement().setProperty("position", position.getPosition());
+    }
+
+    /**
+     * Gets position of the popover with respect to its target.
+     *
+     * @return the position
+     */
+    public PopoverPosition getPosition() {
+        var positionString = getElement().getProperty("position");
+        return Arrays.stream(PopoverPosition.values())
+                .filter(p -> p.getPosition().equals(positionString)).findFirst()
+                .orElse(null);
+    }
+
+    /**
+     * The {@code id} of the element to be used as the popover {@code target}
+     * value.
+     * <p>
+     * The element should be in the DOM by the time when the attribute is set,
+     * otherwise a warning in the Javascript console is shown.
+     *
+     * @param id
+     *            the id of component for this popup, can be {@code null} to
+     *            remove the target
+     */
+    public void setFor(String id) {
+        getElement().setProperty("for", id);
+    }
+
+    /**
+     * Gets the {@code id} of target component of this popup, or {@code null} if
+     * the {@code id} was not set.
+     *
+     * @return the id of target component of this popup
+     * @see #setFor(String)
+     */
+    public String getFor() {
+        return getElement().getProperty("for");
+    }
+
+    /**
+     * Sets the target component for this popover.
+     * <p>
+     * By default, the popover can be opened with a click on the target
+     * component.
+     *
+     * @param target
+     *            the target component for this popover, can be {@code null} to
+     *            remove the target
+     * @throws IllegalArgumentException
+     *             if the target is a {@link Text} component.
+     */
+    public void setTarget(Component target) {
+        if (target instanceof Text) {
+            throw new IllegalArgumentException(
+                    "Text as a target is not supported. "
+                            + "Consider wrapping the Text inside a Div.");
+        }
+
+        this.target = target;
+
+        if (target == null) {
+            getElement().executeJs("this.target = null");
```

Comment:
It seems that the attach listener is currently not removed when the target is disconnected from the popover.

Reply:
Thanks, fixed.